pkg/trace/api/api.go

```diff
@@ -516,13 +522,23 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
 		atomic.AddInt64(&ts.PayloadRefused, 1)
 		return
 	}
+	if err == errHeaderTraceCountNotFound {
+		log.Debugf("Failed to count traces: %s", err)
+	}
+	if err == errHeaderTraceCountNotFound || err == errInvalidHeaderTraceCount {
+		log.Errorf("Failed to count traces: %s", err)
+	}
 
 	start := time.Now()
-	defer func(err error) {
+	defer func() {
 		tags := append(ts.AsTags(), fmt.Sprintf("success:%v", err == nil))
```

Comment:
We do not want to have `success:false` if `traceCount` fails. Its functioning is not essential to success. Thanks for pointing out that after effect as I would've not noticed. Please revert that change.